### PR TITLE
Added Feagin Package

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Checklist

- [X] Appropriate tests were added
- [X] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [X] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

Solves a part of #2177

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "OrdinaryDiffEq"
 uuid = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
 authors = ["Chris Rackauckas <accounts@chrisrackauckas.com>", "Yingbo Ma <mayingbo5@gmail.com>"]
-version = "6.85.0"
+version = "6.84.0"
```
[the remark this answers is not in the document]
Comment (ParamThakkar123) [reply Project.toml]: > this is a mistake, right?

Not sure about that, really.

**Comment by ChrisRackauckas** [comment]
I rebased this to latest master

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas What are the other solvers that we are left to get done with after ?? I will start with them simultaneously and solve the errors from here as well

**Comment by ChrisRackauckas** [comment]
Symplectic RK and Runge Kutta Nystrom (rkn) are probably next. I'll try and do the SDIRK and FIRK, and I think that would just leave Adams methods, BDF, Verner, and Explicit RK. Almost there! Thanks so much

File: lib/OrdinaryDiffEqFeagin/src/OrdinaryDiffEqFeagin.jl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                       calculate_residuals, ssp_coefficient,
+                       OrdinaryDiffEqAlgorithm,
+                       OrdinaryDiffEqMutableCache, OrdinaryDiffEqConstantCache,
+                       OrdinaryDiffEqNewtonAdaptiveAlgorithm,
+                       OrdinaryDiffEqRosenbrockAdaptiveAlgorithm,
+                       OrdinaryDiffEqAdaptiveAlgorithm, CompiledFloats, uses_uprev,
+                       alg_cache, _vec, _reshape, @cache, isfsal, full_cache,
+                       constvalue, _unwrap_val, du_alias_or_new,
+                       explicit_rk_docstring, trivial_limiter!,
+                       _ode_interpolant, _ode_interpolant!,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqFeagin/src/OrdinaryDiffEqFeagin.jl:14]: No custom interpolations here

File: lib/OrdinaryDiffEqFeagin/src/OrdinaryDiffEqFeagin.jl [hunk 3]
```diff
@@ -0,0 +1,28 @@
+module OrdinaryDiffEqFeagin
+
+import OrdinaryDiffEq: alg_order, calculate_residuals!,
+                       initialize!, perform_step!, @unpack, unwrap_alg,
+                       calculate_residuals, ssp_coefficient,
+                       OrdinaryDiffEqAlgorithm,
+                       OrdinaryDiffEqMutableCache, OrdinaryDiffEqConstantCache,
+                       OrdinaryDiffEqNewtonAdaptiveAlgorithm,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqFeagin/src/OrdinaryDiffEqFeagin.jl:8]: No newton algs here

File: lib/OrdinaryDiffEqFeagin/src/OrdinaryDiffEqFeagin.jl [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                       OrdinaryDiffEqRosenbrockAdaptiveAlgorithm,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqFeagin/src/OrdinaryDiffEqFeagin.jl:9]: No resenbrock methods

File: lib/OrdinaryDiffEqFeagin/src/OrdinaryDiffEqFeagin.jl [hunk 5]
```diff
@@ -0,0 +1,28 @@
+module OrdinaryDiffEqFeagin
+
+import OrdinaryDiffEq: alg_order, calculate_residuals!,
+                       initialize!, perform_step!, @unpack, unwrap_alg,
+                       calculate_residuals, ssp_coefficient,
```
Comment (ChrisRackauckas) [inline lib/OrdinaryDiffEqFeagin/src/OrdinaryDiffEqFeagin.jl:5]: No ssp

**Comment by ParamThakkar123** [comment]
@ChrisRackauckas I started a PR for symplectic methods: #2268

**Pull request merged**